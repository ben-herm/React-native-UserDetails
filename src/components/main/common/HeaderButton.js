import React from 'react'
import {StyleSheet, TouchableOpacity, Platform} from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'

class HeaderButton extends React.Component {
  static navigationOptions = ({navigation}) => {
    return {
      title: navigation.getParam('otherParam', 'A Nested Details Screen'),
    }
  }

  render () {
    return (
      <TouchableOpacity
        style={{
          paddingHorizontal: 25,
          marginTop: Platform.OS == 'ios' ? 0 : 20,
        }}
        onPress={() => this.props.navigation.pop()}
        title='Info'
        color='white'>
        <Icon name='arrow-circle-left' size={30} color='#900' />
      </TouchableOpacity>
    )
  }

  /* render function, etc */
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'black',
    flex: 1,
    flexDirection: 'column',
  },
})

export default HeaderButton
