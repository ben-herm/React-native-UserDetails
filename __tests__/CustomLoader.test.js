import React from 'react';
import {CustomLoader} from '../src/components/main/common/CustomLoader';
import renderer from 'react-test-renderer';

describe('renders correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<CustomLoader size={50} />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});
