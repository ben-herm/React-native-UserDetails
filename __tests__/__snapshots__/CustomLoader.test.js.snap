// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 5,
      },
      Object {
        "flex": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(48,118,167)",
        "borderRadius": 50,
        "height": 50,
        "marginHorizontal": 3,
        "transform": Array [
          Object {
            "scale": 0.005,
          },
        ],
        "width": 50,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgb(48,118,167)",
        "borderRadius": 50,
        "height": 50,
        "marginHorizontal": 3,
        "transform": Array [
          Object {
            "scale": 0.005,
          },
        ],
        "width": 50,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgb(48,118,167)",
        "borderRadius": 50,
        "height": 50,
        "marginHorizontal": 3,
        "transform": Array [
          Object {
            "scale": 0.005,
          },
        ],
        "width": 50,
      }
    }
  />
</View>
`;
